### ISSUE TITLE
Cast issue when bumping from 0.9.28 to 0.10.0

### ISSUE THREAD

**Issue by jeanbisutti** [graalvm/native-build-tools#572]
**Describe the bug**

The following issue appears [by bumping to 0.10.0](https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/10383/files) and executing  .`/gradlew nativeTest`:

```
Could not determine the dependencies of task ':instrumentation:logback:logback-appender-1.0:library:nativeTestCompile'.
> class org.graalvm.buildtools.gradle.internal.GraalVMReachabilityMetadataService$Inject cannot be cast to class org.graalvm.buildtools.gradle.internal.GraalVMReachabilityMetadataService (org.graalvm.buildtools.gradle.internal.GraalVMReachabilityMetadataService$Inject is in unnamed module of loader org.gradle.internal.classloader.VisitableURLClassLoader$InstrumentingVisitableURLClassLoader @3fe70938; org.graalvm.buildtools.gradle.internal.GraalVMReachabilityMetadataService is in unnamed module of loader org.gradle.internal.classloader.VisitableURLClassLoader$InstrumentingVisitableURLClassLoader @2ad29845)
```

Native Build Tools configuration: https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/acb078bbc34bb6554ee64725751f2c666bf0724d/instrumentation/logback/logback-appender-1.0/library/build.gradle.kts#L41

Failing job: https://github.com/open-telemetry/opentelemetry-java-instrumentation/actions/runs/7794531614/job/21256023757

**Comment by melix** [comment]
This is a side effect of enabling the metadata repository by default, and a consequence of [this Gradle bug](https://github.com/gradle/gradle/issues/17559).

You can workaround by disabling the metadata repository if you don't need it:

```gradle
graalvmNative {
    metadataRepository {
        enabled.set(false)
    }
}
```

or, you can add the GraalVM plugin to the `buildSrc` classpath instead (see the Gradle issue for details)

**Comment by jeanbisutti** [comment]
@dnestoro I have applied a workaround by disabling the metadata repository.  After doing this, I had to add GraalVM metadata to the Spring Boot project. So, it seems that the workaround has overridden the metadata repository configuration of the Spring Boot Gradle plugin.

I am ok to close this issue. However, by leaving it open, others could be aware of this issue by bumping Native Build Tools to 0.10.0.
